### PR TITLE
feat: add adonis migrations and model for departments

### PR DESCRIPTION
**Pull request by Mikuu-252**
Pull request for migrations and models for departments

### CODE REVIEW

File: database/migrations/1732793568886_create_departments_table.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { BaseSchema } from '@adonisjs/lucid/schema'
+
+export default class extends BaseSchema {
+  protected tableName = 'departments'
+
+  async up() {
+    this.schema.createTable(this.tableName, (table) => {
+      table.increments('id').notNullable()
```
Comment (czaja307) [inline database/migrations/1732793568886_create_departments_table.ts:8]: you don't need to specify `notNullable` on primary key

File: database/migrations/1732793988900_create_fields_of_studies_table.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { BaseSchema } from '@adonisjs/lucid/schema'
+
+export default class extends BaseSchema {
+  protected tableName = 'fields_of_studies'
+
+  async up() {
+    this.schema.createTable(this.tableName, (table) => {
+      table.increments('id').notNullable()
```
Comment (czaja307) [inline database/migrations/1732793988900_create_fields_of_studies_table.ts:8]: same as above

File: database/migrations/1732797273745_create_departments_links_table.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { BaseSchema } from '@adonisjs/lucid/schema'
+import { LinkType } from '../../app/enums/link_type.js'
+
+export default class extends BaseSchema {
+  protected tableName = 'departments_links'
+
+  async up() {
+    this.schema.createTable(this.tableName, (table) => {
+      table.increments('id').notNullable
```
Comment (czaja307) [inline database/migrations/1732797273745_create_departments_links_table.ts:9]: same as above

File: database/migrations/1732797273745_create_departments_links_table.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+      table.text('link').notNullable()
+
+      table.timestamp('created_at').notNullable
+      table.timestamp('updated_at').notNullable
+    })
+  }
+
+  async down() {
+    this.schema.dropTable(this.tableName)
```
Comment (czaja307) [inline database/migrations/1732797273745_create_departments_links_table.ts:29]: you also need to drop the `link_type` enum `if exist`

Comment (czaja307) [reply database/migrations/1732797273745_create_departments_links_table.ts:29]: https://lucid.adonisjs.com/docs/table-builder#enum--enu

File: database/migrations/1732793568886_create_departments_table.ts (hunk 1 above)
Comment (Mikuu-252) [reply database/migrations/1732793568886_create_departments_table.ts]: Changed.

File: database/migrations/1732793988900_create_fields_of_studies_table.ts (hunk 2 above)
Comment (Mikuu-252) [reply database/migrations/1732793988900_create_fields_of_studies_table.ts]: Changed.

File: database/migrations/1732797273745_create_departments_links_table.ts (hunk 3 above)
Comment (Mikuu-252) [reply database/migrations/1732797273745_create_departments_links_table.ts]: Changed.

File: database/migrations/1732797273745_create_departments_links_table.ts (hunk 4 above)
Comment (Mikuu-252) [reply database/migrations/1732797273745_create_departments_links_table.ts:30]: Changed.

**Review by czaja307 — COMMENTED** [review]
i'd be great if you created an `assets` forlder and put your db schema in there for the future reference

**Review by czaja307 — COMMENTED** [review]
wrong pr

File: database/migrations/1732793988900_create_fields_of_studies_table.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { BaseSchema } from '@adonisjs/lucid/schema'
+
+export default class extends BaseSchema {
+  protected tableName = 'fields_of_studies'
+
+  async up() {
+    this.schema.createTable(this.tableName, (table) => {
+      table.increments('id')
+
+      table.integer('department_id').unsigned().references('departments.id').notNullable()
```
[the remark this answers is not in the document]
Comment (Mikuu-252) [reply database/migrations/1732793988900_create_fields_of_studies_table.ts]: Fixed.

File: database/migrations/1732797273745_create_departments_links_table.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import { LinkType } from '../../app/enums/link_type.js'
+
+export default class extends BaseSchema {
+  protected tableName = 'departments_links'
+
+  async up() {
+    this.schema.createTable(this.tableName, (table) => {
+      table.increments('id')
+
+      table.integer('department_id').unsigned().references('departments.id').notNullable()
```
[the remark this answers is not in the document]
Comment (Mikuu-252) [reply database/migrations/1732797273745_create_departments_links_table.ts]: Fixed.

**Review by czaja307 — APPROVED** [review]
LGTM

**Pull request merged**